jupyterhub/templates/proxy/deployment.yaml

```diff
@@ -1,40 +1,127 @@
         checksum/secret: {{ include (print $.Template.BasePath "/hub/secret.yaml") . | sha256sum }}
+      labels:
+        name: proxy
+        component: proxy
+        release: {{ .Release.Name }}
+        heritage: {{ .Release.Service }}
+        # required for kube-lego to work
+        app: kube-lego
     spec:
+      {{- if .Values.proxy.rbac.enabled }}
+      serviceAccountName: proxy
+      {{- end }}
       containers:
-      - name: proxy-container
-        image: {{ .Values.proxy.image.name }}:{{ .Values.proxy.image.tag }}
-        {{- if .Values.proxy.cmd }}
-        command: {{ toJson .Values.proxy.cmd }}
+        - name: nginx
+          image: "{{ .Values.proxy.nginx.image.name }}:{{ .Values.proxy.nginx.image.tag }}"
+          imagePullPolicy: {{ .Values.proxy.nginx.image.pullPolicy }}
+          resources:
+{{ toYaml .Values.proxy.nginx.resources | indent 12 }}
+          args:
+            - /nginx-ingress-controller
+            - --default-backend-service={{ .Release.Namespace }}/proxy-http
+            - --ingress-class=jupyterhub-proxy-tls
+            - --watch-namespace={{ .Release.Namespace }}
+          env:
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: POD_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+          livenessProbe:
+            httpGet:
+              path: /healthz
+              port: 10254
+              scheme: HTTP
+            initialDelaySeconds: 10
+            timeoutSeconds: 1
+          readinessProbe:
+            httpGet:
+              path: /healthz
+              port: 10254
+              scheme: HTTP
+          ports:
+            - name: http
+              containerPort: 80
+              protocol: TCP
+            - name: https
+              containerPort: 443
+              protocol: TCP
+        {{ if .Values.proxy.hosts -}}
+        - name: kube-lego
+          image: "{{ .Values.proxy.lego.image.name }}:{{ .Values.proxy.lego.image.tag }}"
+          imagePullPolicy: {{ .Values.proxy.lego.image.pullPolicy }}
+          resources:
+{{ toYaml .Values.proxy.lego.resources | indent 12 }}
+          env:
+            - name: LEGO_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+            - name: LEGO_WATCH_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+            - name: LEGO_POD_IP
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.podIP
+            - name: LEGO_EMAIL
+              # {{ required "proxy.https.auto.letsencryptContactEmail is a required field" .Values.proxy.https.auto.letsencryptContactEmail }}
+              value: {{ .Values.proxy.https.auto.letsencryptContactEmail | quote }}
```

Comment:
The field `letsencryptContactEmail` doesn't quite match the docs: `letsEncryptContactEmail`